### PR TITLE
HTML5 range documentation

### PR DESCRIPTION
**Pull request by harikt**
Hi, 

| Q | A |
| --- | --- |
| Doc fix? | yes |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | [#5439] |
| Code PR | [12607] |
| License | MIT |

I am trying to contribute to range added for the PR https://github.com/symfony/symfony/pull/12067 & https://github.com/symfony/symfony/issues/11979 which fixes https://github.com/symfony/symfony-docs/issues/5439 .

Need some help on moving for I am really new to the documentation. 

The current issues I feel is whether we need to add Basic usage or is that only reference ?

I had a look at http://symfony.com/doc/current/reference/forms/types/number.html which would have been similar, but that seems it referencing to choice seems more confusing.

I have looked into other types also and almost all are some what similar.

Any input is highly appreciated and will work on in the free time. If someone is interested please do take the PR or finish it .

Thank you.

### CODE REVIEW

File: reference/forms/types/range.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
++-------------+---------------------------------------------------------------------+
+| Inherited   | - `data`_                                                           |
+| options     | - `disabled`_                                                       |
+|             | - `empty_data`_                                                     |
+|             | - `error_bubbling`_                                                 |
+|             | - `error_mapping`_                                                  |
+|             | - `label`_                                                          |
+|             | - `label_attr`_                                                     |
+|             | - `mapped`_                                                         |
+|             | - `max`_                                                           |
```
Comment (harikt) [inline reference/forms/types/range.rst]: Hm. Anyone have idea how many spaces should be kept ?

Comment (xabbuh) [inline reference/forms/types/range.rst]: You need one more in this and in the next line (have a look at the right most `|` character).

File: reference/forms/types/range.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+   single: Forms; Fields; range
+
+range Field Type
+================
+
+The ``range`` field is a slider that is rendered using the HTML5
+``<input type="range" />`` tag.
+
++-------------+---------------------------------------------------------------------+
+| Rendered as | ``input`` ``range`` field (slider in html5 supported browser)                              |
```
Comment (xabbuh) [inline reference/forms/types/range.rst]: HTML5

Comment (xabbuh) [inline reference/forms/types/range.rst]: and you need to remove some spaces

File: reference/forms/types/range.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
++-------------+---------------------------------------------------------------------+
+| Parent type | :doc:`text </reference/forms/types/text>`                           |
++-------------+---------------------------------------------------------------------+
+| Class       | :class:`Symfony\\Component\\Form\\Extension\\Core\\Type\\RangeType` |
++-------------+---------------------------------------------------------------------+
+
+Basic Usage
+-----------
+
+
```
Comment (xabbuh) [inline reference/forms/types/range.rst]: Please remove one of the blank lines.

File: reference/forms/types/range.rst (hunk 1 above)
Comment (xabbuh) [inline reference/forms/types/range.rst]: By the way, `min` and `max` are no options of the form type. You need to pass them as `attr` option keys.

**Comment by harikt** [comment]
Hey @xabbuh , 

Thanks for all the comments. What about the examples ?

**Comment by xabbuh** [comment]
I think a simple example showing how to configure the `min` and `max` attributes should be enough (see, for example, the test cases in the code pull request: https://github.com/symfony/symfony/pull/12067/files#diff-e11656489f1e01413320bd5edff0f0c7R1350).

**Comment by harikt** [comment]
Hey @xabbuh , 

May be I didn't explained better regarding  what I was referencing to.

Basically if you check the markdown of http://symfony.com/doc/current/reference/forms/types/number.html ( https://raw.githubusercontent.com/harikt/symfony-docs/range/reference/forms/types/number.rst ) , you can see the `number` was not having the correct examples. It refers to `hidden`, `choice` fields etc for `data`, `empty_data` etc .

I will make the changes and get back.

Thank you.

File: reference/forms/types/range.rst (hunk 1 above)
Comment (harikt) [inline reference/forms/types/range.rst]: may be we need `attr` here.

File: reference/forms/types/range.rst (hunk 2 above)
Comment (harikt) [inline reference/forms/types/range.rst]: Basically the editor was having `Soft wrap at preferred length` turned on . I changed it. Thanks.

File: reference/forms/types/range.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
++-------------+---------------------------------------------------------------------+
+| Parent type | :doc:`text </reference/forms/types/text>`                           |
++-------------+---------------------------------------------------------------------+
+| Class       | :class:`Symfony\\Component\\Form\\Extension\\Core\\Type\\RangeType` |
++-------------+---------------------------------------------------------------------+
+
+Basic Usage
+-----------
+
+    $builder->add('name', 'range', array(
```
Comment (snoek09) [inline reference/forms/types/range.rst]: Please add a code-block directive before the example code:

```rst
.. code-block:: php

    $builder->add('name', 'range', array(
```

File: reference/forms/types/range.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+range Field Type
+================
+
+The ``range`` field is a slider that is rendered using the HTML5
+``<input type="range" />`` tag.
+
++-------------+---------------------------------------------------------------------+
+| Rendered as | ``input`` ``range`` field (slider in HTML5 supported browser)       |
++-------------+---------------------------------------------------------------------+
+| Inherited   | - `data`_                                                           |
```
Comment (snoek09) [inline reference/forms/types/range.rst]: Please add missing include for `data` option.

File: reference/forms/types/range.rst [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+| options     | - `disabled`_                                                       |
+|             | - `empty_data`_                                                     |
```
Comment (snoek09) [inline reference/forms/types/range.rst]: Please add missing include for `empty_data` option.

File: reference/forms/types/range.rst [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+.. include:: /reference/forms/types/options/error_mapping.rst.inc
+
+.. include:: /reference/forms/types/options/label.rst.inc
+
+.. include:: /reference/forms/types/options/label_attr.rst.inc
+
+.. include:: /reference/forms/types/options/mapped.rst.inc
+
+.. include:: /reference/forms/types/options/read_only.rst.inc
```
Comment (snoek09) [inline reference/forms/types/range.rst]: Please add `read_only` to inherited options list.

**Comment by snoek09** [comment]
The new range documentation isn't included anywhere yet.
Please add it to the list in `/reference/forms/types.rst`

I found these small issues by testing the documentation first on my local machine:
http://symfony.com/doc/current/contributing/documentation/format.html#testing-documentation

File: reference/forms/types/range.rst (hunk 6 above)
Comment (harikt) [inline reference/forms/types/range.rst]: Hey @snoek09 actually I have a question regarding the inherited options and include. I actually copied this from `email.rst` or something I guess. So is it all the properties of the text field for it is parent the properties are imported ?

File: reference/forms/types/range.rst (hunk 4 above)
Comment (harikt) [inline reference/forms/types/range.rst]: sure. Thanks.

File: reference/forms/types/range.rst (hunk 6 above)
Comment (snoek09) [inline reference/forms/types/range.rst]: `EmailType` and `RangeType` both have the same type as a parent; **text** (`TextType`).

The parent type of **text** is **form** (`FormType`). The inherited options are from this **form** type:

http://symfony.com/doc/current/reference/forms/types/text.html

Comment (harikt) [inline reference/forms/types/range.rst]: cool.

Thank you.

File: reference/forms/types/range.rst (hunk 7 above)
Comment (harikt) [inline reference/forms/types/range.rst]: - `read_only`_ (deprecated as of 2.8) , do we need to add ? May be remove it from docs ?

**Comment by harikt** [comment]
@xabbuh @snoek09 thank you guys for your time and review.

Have updated the changes.

Thank you.

File: reference/forms/types/range.rst [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+.. include:: /reference/forms/types/options/error_bubbling.rst.inc
+
+.. include:: /reference/forms/types/options/error_mapping.rst.inc
+
+.. include:: /reference/forms/types/options/label.rst.inc
+
+.. include:: /reference/forms/types/options/mapped.rst.inc
+
+.. include:: /reference/forms/types/options/label_attr.rst.inc
```
Comment (xabbuh) [inline reference/forms/types/range.rst]: This description should be moved before the `mapped` option's description.

**Comment by xabbuh** [comment]
Very great @harikt. I left you one minor comment. After that I think this looks ready to be merged.

**Comment by harikt** [comment]
@xabbuh done :+1:  .

Thank you.

**Comment by xabbuh** [comment]
:+1:

**Comment by weaverryan** [comment]
@harikt! This is wonderful! I have no changes - and now I can also close the related issue :). Thanks!

**Comment by harikt** [comment]
cool. Thank you all.

**Pull request merged**